dtls.c

```diff
@@ -1539,6 +1510,98 @@ dtls_prepare_record(dtls_peer_t *peer, dtls_security_parameters_t *security,
   return 0;
 }
 
+/**
+ * Send Alert in stateless fashion.
+ * A Alert is sent to the peer (using the write callback function
+ * registered with \p ctx). The return value is \c -1 on error, \c 1 when
+ * the Alert is sent.
+ *
+ * \param ctx              The DTLS context.
+ * \param ephemeral_peer   The ephemeral remote party we are talking to.
+ * \param level            Alert level.
+ * \param description      Alert description.
+ * \return \c 1 if msg is sent, or \c -1 otherwise.
+ */
+static int
+dtls_0_send_alert(dtls_context_t *ctx,
+			     dtls_ephemeral_peer_t *ephemeral_peer,
+			     dtls_alert_level_t level,
+			     dtls_alert_t description)
+{
+  uint8 buf[DTLS_RH_LENGTH + DTLS_ALERT_LENGTH];
+  uint8 *p = dtls_set_record_header(DTLS_CT_HANDSHAKE, 0, &(ephemeral_peer->rseq), buf);
+
+  /* fix length of fragment in sendbuf */
+  dtls_int_to_uint16(buf + 11, DTLS_ALERT_LENGTH);
+
+  /* Alert */
+  dtls_int_to_uint8(p, level);
+  dtls_int_to_uint8(p + 1, description);
+
+  dtls_debug("send alert - protocol version  packet\n");
+
+  dtls_debug_hexdump("send header", buf, DTLS_RH_LENGTH);
+  dtls_debug_hexdump("send unencrypted alert", p, DTLS_ALERT_LENGTH);
+
+  int res = CALL(ctx, write, ephemeral_peer->session, buf, sizeof(buf));
+
+  /* Guess number of bytes application data actually sent:
+   * dtls_prepare_record() tells us in len the number of bytes to
+   * send, res will contain the bytes actually sent. */
+  return res <= 0 ? res : (int) sizeof(buf);
```

Comment:
It is unclear to me why `res` is not passed through, especially since `sizeof(buf)` ≠ 1 (c.f. doxygen comment for `\return`).

Reply:
Copied from other functions :-). 
Replaced by res.